### DISCUSSION TITLE
the use of 'self.embed' in deepVARNetwork

### DISCUSSION THREAD

**Discussion by ShirleyMgit** [awslabs/gluonts#2720] [Q&A]
Hi,
I am trying to extend the network with an embedding layer for static features (similarly to deepAR).
Therefore, I am trying to understand the code. 
Can you please explain what is the role of 'self.embed' and repeated_index_embeddings?
thanks

**Comment by mbohlkeschneider** [comment]
Hi @ShirleyMgit,

Thank you for your question. The embedding that is implemented isn't really useful for `DeepVAR` but is used in `CPVAR` ([code](https://github.com/awslabs/gluonts/blob/dev/src/gluonts/mx/model/gpvar/_network.py#L25)), which is derived from `DeepVAR`. Here, we sample the dimensions of the multivariate dataset and project into the low-rank covariance matrix (see this [paper](https://proceedings.neurips.cc/paper/2019/hash/0b105cf1504c4e241fcc6d519ea962fb-Abstract.html)). The reason it is there for `DeepVAR` is merely to reduce code duplication for `CPVAR`. Hope that was clear.
